### PR TITLE
Bugfix/#682

### PR DESCRIPTION
**Pull request by manoramahp**
Fix for https://github.com/Cadasta/cadasta-platform/issues/682

Addressed the issues mentioned in the comment in https://github.com/Cadasta/cadasta-platform/pull/702

Please review and merge

### CODE REVIEW

**Comment by seav** [comment]
28 tests are now failing.

**Comment by ian-ross** [comment]
@manoramahp Can you run tests locally and make sure things pass before pushing to GitHub please?

**Comment by ian-ross** [comment]
@manoramahp For the second time of asking, please run and fix tests locally before pushing to GitHub.  Every time you push to an open PR, all of our tests run on Travis.  That's just a waste of resources and time.  It's much quicker for you to run tests locally to find problems.  You can run *all* of the tests (PEP8 checking, unit tests and functional tests) using the `tox` command, or you can just run unit tests directly using `py.test` (https://devwiki.corp.cadasta.org/Running%20tests%20for%20the%20back%20end).

File: cadasta/organization/tests/test_views_default_projects.py [hunk 1]
```diff
@@ -219,10 +219,11 @@ def test_get_with_superuser(self):
         self.user.assign_policies(superuser_role)
         response = self.request(user=self.user)
         assert response.status_code == 200
-        assert response.content == self.render_content(is_superuser=True,
-                                                       is_administrator=True,
-                                                       is_allowed_add_location=True,
-                                                       is_allowed_add_resource=True)
+        assert response.content == self\
```
Comment (ian-ross) [inline cadasta/organization/tests/test_views_default_projects.py]: Please don't use end-of-line backslashes like this.  You can see plenty of examples of more readable ways to break long lines in other parts of the code.

File: cadasta/organization/tests/test_views_default_projects.py [hunk 2]
```diff
@@ -324,10 +325,11 @@ def test_get_private_project_with_superuser(self):
...
+            .render_content(is_superuser=True,
+                               is_administrator=True,
+                                    is_allowed_add_location=True,
+                                        is_allowed_add_resource=True)
```
Comment (ian-ross) [inline cadasta/organization/tests/test_views_default_projects.py]: All of the test changes you've made have `is_allowed_add_location` and `is_allowed_add_resource` set to `True`.  You should write some tests that have different values for those flags to check that the rendering is correct in some other cases.

**Comment by ian-ross** [comment]
@manoramahp You are still pushing changes to your branch before ensuring that tests pass.  Every single time you do this, all of the tests run on Travis and fail.  Do you not know how to run tests locally?

**Comment by manoramahp** [comment]
@ian-ross Apologies for the inconvenience. The last commit was done after testing locally. But I haven't get a pull from master branch so one newly added test has failed. Now I've fixed the test failure.

File: cadasta/organization/tests/test_views_default_projects.py (hunk 1 above)
Comment (manoramahp) [inline cadasta/organization/tests/test_views_default_projects.py]: Corrections done

File: cadasta/organization/tests/test_views_default_projects.py (hunk 2 above)
Comment (manoramahp) [inline cadasta/organization/tests/test_views_default_projects.py]: Added different values for flags

File: cadasta/organization/views/mixins.py [hunk 3]
```diff
@@ -127,6 +127,10 @@ def is_administrator(self):
     def get_context_data(self, *args, **kwargs):
         context = super().get_context_data(*args, **kwargs)
         context['is_administrator'] = self.is_administrator
+        context['is_allowed_add_location'] = self\
```
Comment (oliverroick) [inline cadasta/organization/views/mixins.py]: Please do not use backslashes for line breaks. You could instead store the user in a local variable and add line breaks in-between method parameters:

```python
user = self.request.user
context['is_allowed_add_location'] = user.has_perm(
    'spatial.create',
    self.get_project()
)
```

File: cadasta/organization/tests/test_views_default_projects.py [hunk 4]
```diff
@@ -223,20 +223,33 @@ def setup_url_kwargs(self):
...
 
     def test_get_with_superuser(self):
         superuser_role = Role.objects.get(name='superuser')
         self.user.assign_policies(superuser_role)
         response = self.request(user=self.user)
         assert response.status_code == 200
-        assert response.content == self.render_content(is_superuser=True,
-                                                       is_administrator=True)
+        assert response.content == self.render_content(
+                            is_superuser=True,
```
Comment (oliverroick) [inline cadasta/organization/tests/test_views_default_projects.py]: Can you make the indentation consistent with all other tests, please?

File: cadasta/organization/tests/test_views_default_projects.py [hunk 5]
```diff
@@ -333,8 +364,11 @@ def test_get_private_project_with_superuser(self):
         self.user.assign_policies(self.superuser_role)
         response = self.request(user=self.user)
         assert response.status_code == 200
-        assert response.content == self.render_content(is_superuser=True,
-                                                       is_administrator=True)
+        assert response.content == self.render_content(
+                            is_superuser=True,
```
Comment (oliverroick) [inline cadasta/organization/tests/test_views_default_projects.py]: Can you make the indentation consistent with all other tests, please?

File: cadasta/organization/views/mixins.py (hunk 3 above)
Comment (manoramahp) [inline cadasta/organization/views/mixins.py]: @oliverroick Thanks for the feedback. I've done the required fixes. https://github.com/Cadasta/cadasta-platform/pull/703/commits/ef8577cc0ee6c2bda85f25ada9fc983ec826def0#diff-e68c62586856b6f7b38d7f4629f0ce4bR130

File: cadasta/organization/tests/test_views_default_projects.py (hunk 5 above)
Comment (manoramahp) [inline cadasta/organization/tests/test_views_default_projects.py]: Done the code formatting fixes in https://github.com/Cadasta/cadasta-platform/pull/703/commits/ef8577cc0ee6c2bda85f25ada9fc983ec826def0

File: cadasta/templates/organization/project_dashboard.html [hunk 6]
```diff
@@ -106,11 +106,13 @@
           <h3 style="text-transform: capitalize;">{% trans "Congratulations!" %}</h3>
           <p>{% trans "You have successfully created your project.  You're now ready to add your first location." %}</p>
         {% endif %}
+        {% if is_allowed_add_resource %}
```
Comment (oliverroick) [inline cadasta/templates/organization/project_dashboard.html]: Shouldn't you be checking for `is_allowed_location` here?

File: cadasta/templates/organization/project_dashboard.html [hunk 7]
```diff
@@ -76,7 +76,7 @@
           </dl>
         {% endif %}
         <div class="divider-thick"></div>
-        {% if has_content %}
+        {% if has_content or not is_allowed_add_resource %}
```
Comment (oliverroick) [inline cadasta/templates/organization/project_dashboard.html]: Sorry, I might miss something here, but I don't understand this part. Why are the stats displayed when the user is not allowed to add resources?

File: cadasta/templates/organization/project_dashboard.html (hunk 6 above)
Comment (manoramahp) [inline cadasta/templates/organization/project_dashboard.html]: Yes, that should be  `is_allowed_location` . Thanks for pointing it out. Fixed it in https://github.com/Cadasta/cadasta-platform/pull/703/commits/8b45ed8b5197b3747a4aba8a50284aaffdb73fff

**Comment by oliverroick** [comment]
Hi @manoramahp, thanks for fixing the PR. That's looking good now.

I have one final request, before we can merge your PR. Could you please squash the 23 commits in your branch into one and then rebase onto the latest changes in master? That will make merging your changes a lot easier for us. 

You'll see that GitHub says "This branch is out-of-date with the base branch".  That means that there have been changes in the main `Cadasta/cadasta-platform` repository that you don't have in your PR branch.  That means that there's no way to know if your work is compatible with what's already been done.  What you need to do in this case is to make sure that you've added our repo as an `upstream` remote for your fork (GitHub usually tells you to do this when you create a fork), fetch the latest `master` from there, and then update your PR branch to incorporate the latest changes.  (This is the point where working directly on `master` in your repo turns out to be a bad idea...)  If you're working on a topic branch, you do this:

```
git checkout master  # This is your local master branch
git fetch upstream/master  # Get changes from Cadasta/cadasta-platform
git merge upstream/master  # This does a "fast-forward" merge
git checkout <topic-branch>  # Switch back to your topic branch
git rebase master  # Replay the changes you've made on top of the latest upstream master
git push --force  # Update your PR on GitHub
```

If there are conflicts between your work and what's been done in the main repo, after you do the rebase, you'll get merge conflicts that you need to fix.

**Comment by manoramahp** [comment]
@oliverroick Added the changes to one commit. https://github.com/Cadasta/cadasta-platform/pull/736

**Pull request closed without merging**